### PR TITLE
update mysql statements within core to make VMangos compatible with newer MySQL versions

### PR DESCRIPTION
**Pull request by schell244**
inspired by Mangos Zero https://github.com/mangoszero/server/commit/7fd24f7a6ae4345608087a433b4662cfe6b6fa61

### CODE REVIEW

File: src/game/Commands/AccountCommands.cpp [hunk 1]
```diff
@@ -553,7 +553,7 @@ bool ChatHandler::HandleBanAllIPCommand(char* args)
     } while (result->NextRow());
 
     delete result;
-    if (result = CharacterDatabase.PQuery("SELECT `account` FROM `characters` WHERE `account` IN '%s' AND `level` > '%u' GROUP BY `account`", allAccounts.str().c_str(), maxLevel))
+    if (result = CharacterDatabase.PQuery("SELECT `account` FROM `characters` WHERE `account` IN (`%s`) AND `level` > '%u' GROUP BY `account`", allAccounts.str().c_str(), maxLevel))
```
Comment (ratkosrb) [inline src/game/Commands/AccountCommands.cpp]: Bad change, you want '%s'.

Comment (schell244) [reply src/game/Commands/AccountCommands.cpp]: fixed with https://github.com/vmangos/core/pull/896/commits/4b5b0bf7ff8c5f6fd7abea694d2105fb8ddb9c8c

**Comment by ulfgebhardt** [comment]
@ratkosrb can we go through with this? <3

**Comment by ratkosrb** [comment]
I have to look through all the changes first.

File: src/game/Commands/AccountCommands.cpp [hunk 2]
```diff
@@ -809,7 +809,7 @@ bool ChatHandler::HandleBanInfoIPCommand(char* args)
     std::string IP = cIP;
 
     LoginDatabase.escape_string(IP);
-    QueryResult* result = LoginDatabase.PQuery("SELECT ip, FROM_UNIXTIME(bandate), FROM_UNIXTIME(unbandate), unbandate-UNIX_TIMESTAMP(), banreason,bannedby,unbandate-bandate FROM ip_banned WHERE ip = '%s'", IP.c_str());
+    QueryResult* result = LoginDatabase.PQuery("SELECT `ip`, FROM_UNIXTIME(`bandate`), FROM_UNIXTIME(`unbandate`), `unbandate`-UNIX_TIMESTAMP(), `banreason`, `bannedby`, `unbandate-bandate` FROM `ip_banned` WHERE `ip` = '%s'", IP.c_str());
```
Comment (ratkosrb) [inline src/game/Commands/AccountCommands.cpp:812]: unbandate-bandate is not a column, those are 2 columns being substracted

File: src/game/Commands/CharacterCommands.cpp [hunk 3]
```diff
@@ -883,7 +883,7 @@ bool ChatHandler::HandleRemoveRidingCommand(char* args)
             // Journeyman Riding
             case 33391:
             {
-                if (!CharacterDatabase.PExecute("UPDATE character_skills SET value = 75, max = 75 WHERE skill = 762 AND value = 150 AND max = 150 AND guid = %u", target_guid.GetCounter()))
+                if (!CharacterDatabase.PExecute("UPDATE `character_skills` SET `value` = 75, `max` = 75 WHERE `skill` = 762 AND `value` = 150 AND `max` = `150` AND `guid` = '%u'", target_guid.GetCounter()))
```
Comment (ratkosrb) [inline src/game/Commands/CharacterCommands.cpp]: 150 is not a column or table name

File: src/game/Commands/CharacterCommands.cpp [hunk 4]
```diff
@@ -1255,12 +1255,12 @@ bool ChatHandler::GetDeletedCharacterInfoList(DeletedInfoList& foundList, bool u
                 std::copy(list.begin(), list.end(), std::ostream_iterator<int>(accountStream, ","));
                 std::string accounts = accountStream.str();
                 accounts.pop_back();
-                resultChar = CharacterDatabase.PQuery("SELECT guid, deleteInfos_Name, deleteInfos_Account, deleteDate FROM characters WHERE deleteDate IS NOT NULL AND deleteInfos_Account IN (%s) LIMIT 0,50", accounts.c_str());
+                resultChar = CharacterDatabase.PQuery("SELECT `guid`, `deleteInfos_Name`, `deleteInfos_Account`, `deleteDate` FROM `characters` WHERE `deleteDate` IS NOT NULL AND `deleteInfos_Account` IN '%s' LIMIT 0,50", accounts.c_str());
```
Comment (ratkosrb) [inline src/game/Commands/CharacterCommands.cpp]: what are you doing here, why did u remove the brackets around %s, this is where the list will go

File: src/game/Commands/AccountCommands.cpp (hunk 2 above)
Comment (schell244) [reply src/game/Commands/AccountCommands.cpp:812]: https://github.com/vmangos/core/pull/896/commits/c552f3bdd708ebbee45b02b1a864d228c4f6dde0

File: src/game/Commands/CharacterCommands.cpp (hunk 3 above)
Comment (schell244) [reply src/game/Commands/CharacterCommands.cpp]: fixed with https://github.com/vmangos/core/pull/896/commits/c552f3bdd708ebbee45b02b1a864d228c4f6dde0

File: src/game/Commands/CharacterCommands.cpp (hunk 4 above)
Comment (schell244) [reply src/game/Commands/CharacterCommands.cpp]: fixed with https://github.com/vmangos/core/pull/896/commits/c552f3bdd708ebbee45b02b1a864d228c4f6dde0

**Comment by schell244** [comment]
Closing this, becasue this becasue it's to risky to merge such big changes at once.

**Comment by ulfgebhardt** [comment]
Could you open an Issue instead so we don't lose track on it @schell244 ?

**Comment by schell244** [comment]
I've just talked with @ratkosrb and he told me that he's working on a PR where mysql keywords are escaped instead of all queries. Makes more sense, and is safer. So it's not out of track. I could still open a ticket to have it documented somewhere?

**Comment by ulfgebhardt** [comment]
I you would, please open a ticket - as soon as its done we can close it @schell244 <3

**Pull request closed without merging**